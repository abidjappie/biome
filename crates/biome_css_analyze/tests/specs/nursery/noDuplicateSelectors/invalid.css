/* selectors using parts of other selectors */
.goo .gar {
    div {
        s {
            
        }
    }
    z {
        m {

        }
        n {
            
        }
    }
}

.goo {
    div {
        s {

        }
    }
}

/* TODO: this case does not work yet because it doesn't compare on root level */
.goo div {}