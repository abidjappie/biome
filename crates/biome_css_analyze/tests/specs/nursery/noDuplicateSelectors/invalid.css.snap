---
source: crates/biome_css_analyze/tests/spec_tests.rs
expression: invalid.css
---
# Input
```css
/* selectors using parts of other selectors */
.goo .gar {
    div {
        s {
            
        }
    }
    z {
        m {

        }
        n {
            
        }
    }
}

.goo {
    div {
        s {

        }
    }
}

/* TODO: this case does not work yet because it doesn't compare on root level */
.goo div {}
```

# Diagnostics
```
invalid.css:3:5 lint/nursery/noDuplicateSelectors ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Duplicate selectors may result in unintentionally overriding rules.
  
    1 │ /* selectors using parts of other selectors */
    2 │ .goo .gar {
  > 3 │     div {
      │     ^^^
    4 │         s {
    5 │             
  
  i Please consider moving the rule's contents to the first occurence:
  
    18 │ .goo {
  > 19 │     div {
       │     ^^^
    20 │         s {
    21 │ 
  
  i Remove duplicate selectors within the rule
  

```

```
invalid.css:4:9 lint/nursery/noDuplicateSelectors ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ! Duplicate selectors may result in unintentionally overriding rules.
  
    2 │ .goo .gar {
    3 │     div {
  > 4 │         s {
      │         ^
    5 │             
    6 │         }
  
  i Please consider moving the rule's contents to the first occurence:
  
    18 │ .goo {
    19 │     div {
  > 20 │         s {
       │         ^
    21 │ 
    22 │         }
  
  i Remove duplicate selectors within the rule
  

```
